tests/v2/integration/catalogv2/system_charts_version_test.go

```diff
@@ -0,0 +1,272 @@
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	stevev1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/clients/rancher/catalog"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+)
+
+type SystemChartsVersionSuite struct {
+	suite.Suite
+	client               *rancher.Client
+	session              *session.Session
+	restClientGetter     genericclioptions.RESTClientGetter
+	catalogClient        *catalog.Client
+	latestWebhookVersion string
+}
+
+func (w *SystemChartsVersionSuite) TearDownSuite() {
+	_ = w.updateSetting("rancher-webhook-version", w.latestWebhookVersion)
+	_ = w.updateSetting("rancher-webhook-min-version", "")
+	_ = w.updateSetting("system-feature-chart-refresh-seconds", "3600")
+	w.session.Cleanup()
+}
+
+func (w *SystemChartsVersionSuite) SetupSuite() {
+	var err error
+	testSession := session.NewSession()
+	w.session = testSession
+
+	w.client, err = rancher.NewClient("", testSession)
+	require.NoError(w.T(), err)
+
+	w.catalogClient, err = w.client.GetClusterCatalogClient("local")
+	require.NoError(w.T(), err)
+
+	host := hostName(w.client.RestConfig.Host)
+	w.restClientGetter = &genericclioptions.ConfigFlags{
+		CAFile:      &w.client.RestConfig.TLSClientConfig.CAFile,
+		APIServer:   &host,
+		Insecure:    &w.client.RestConfig.TLSClientConfig.Insecure,
+		BearerToken: &w.client.RestConfig.BearerToken,
+	}
+
+	w.latestWebhookVersion, err = w.catalogClient.GetLatestChartVersion("rancher-webhook")
+	require.NoError(w.T(), err)
+
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-version", w.latestWebhookVersion))
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-min-version", ""))
+	require.NoError(w.T(), w.updateSetting("system-feature-chart-refresh-seconds", "10"))
+}
+
+func (w *SystemChartsVersionSuite) resetSettings() {
+	w.T().Helper()
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-version", w.latestWebhookVersion))
+	require.NoError(w.T(), w.updateSetting("rancher-webhook-min-version", ""))
+	require.NoError(w.T(), w.updateSetting("system-feature-chart-refresh-seconds", "10"))
+}
+
+func TestSystemChartsVersionSuite(t *testing.T) {
+	suite.Run(t, new(SystemChartsVersionSuite))
+}
+
+func (w *SystemChartsVersionSuite) TestInstallWebhook() {
+	defer w.resetSettings()
+
+	const exactVersion = "2.0.3+up0.3.3"
```

Comment:
should we put consts at the top of the file.

Reply:
I don't think so, since they are scoped to individual tests.